AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt

```diff
@@ -247,11 +247,22 @@ object ImportUtils {
 
         protected open fun getFileNameFromContentProvider(context: Context, data: Uri): String? {
             var filename: String? = null
-            context.contentResolver.query(data, arrayOf(OpenableColumns.DISPLAY_NAME), null, null, null).use { cursor ->
-                if (cursor != null && cursor.moveToFirst()) {
-                    filename = cursor.getString(0)
-                    Timber.d("handleFileImport() Importing from content provider: %s", filename)
+            try {
+                context.contentResolver.query(
+                    data,
+                    arrayOf(OpenableColumns.DISPLAY_NAME),
+                    null,
+                    null,
+                    null
+                ).use { cursor ->
+                    if (cursor != null && cursor.moveToFirst()) {
+                        filename = cursor.getString(0)
+                        Timber.d("handleFileImport() Importing from content provider: %s", filename)
+                    }
                 }
+            } catch (e: Exception) {
+                Timber.e(e, "Error querying content provider")
```

Comment:
One nitpick here - this is unexpected and serious (it would have crashed!) but we can recover.
That's our warning level, not error: https://github.com/ankidroid/Anki-Android/wiki/Code-style#log-levels

```suggestion
                Timber.w(e, "Error querying content provider")
```

```
The ERROR level should only be used when something fatal has happened, and it is not recoverable
The WARNING level should be used when something serious and unexpected happened, but you can recover
The INFORMATIVE level should be used to note something interesting to most people
The DEBUG level should be used to note what is happening on the device that could be relevant to investigate and debug unexpected behaviours.
The VERBOSE level should be used for everything else.
A DEBUG log should be inside an if (LOCAL_LOGD) block.
```